sktime/forecasting/croston.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import pandas as pd
+from sktime.forecasting.base._sktime import _SktimeForecaster
+from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+
+
+DEFAULT_ALPHA = 0.05
+
+
+class Croston(_OptionalForecastingHorizonMixin, _SktimeForecaster):
+    """
+    Implementation of Croston's Method
+    ----------------------------------
+    Useful for Forecasting Intermittent Demand Time Series.
+    The Croston() function produces forecasts using Croston’s method.
+    It simply uses α = 0.1 by default,
+    and p = 0 is set to be equal to the first observation in each of the series.
+    This is consistent with the way Croston envisaged the method being used.
+
+    Parameters:
+    -----------
+        demand: array-like
+            Historical data
+        future_periods: int
+            Time period for which predictions are required
+        alpha: float, optional(default=0.1)
+            Smoothing parameter
+
+    Returns:
+    --------
+        forecast: array-like
+            Forecasted demand (on average per period) diff."""
+
+    def __init__(self, alpha=DEFAULT_ALPHA):
+        # hyperparameter
+        self.alpha = alpha
+
+        # training data
+        self._y = None
+        self._X = None
+
+        # forecasting horizon
+        self._fh = None
+        # self.cutoff = None  # reference point for relative fh
+
+        # set _is_fitted to False
+        self._is_fitted = False
+        super(Croston, self).__init__()
+
+    def fit(self, y, X=None, fh=None):
+        if X is not None:
+            raise NotImplementedError(
+                "Support for exogenous variables is not yet implemented"
+            )
+        self._set_y_X(y, X)
+        self._set_fh(fh)
+        self._is_fitted = True
+
+        return self
+
+    def predict(self, fh=None, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        if return_pred_int or X is not None:
+            raise NotImplementedError()
+        self.check_is_fitted()
+        self._set_fh(fh)
+
+        future_periods = len(self._fh.to_numpy())
```

Comment:
My impression is that some of the variable naming makes it hard to read. For example:
* `cols` is what we usually call `n_timepoints`, 
* `future_periods` is `len_fh`,
* `d` is renamed, first it's just `y`, then it's the entire series with preallocated space for the predictions